File: leetcode/longest-common-prefix.java
//https://leetcode.com/problems/longest-common-prefix/

class Solution {
    public String longestCommonPrefix(String[] strs) {
        String prefix = "";
        Boolean processing = true;
        int index = 0;
        
        while(processing){
            for(String i:strs){
                if(i.length()-1<index) return prefix.substring(0,index);
                if(prefix.length()-1<index) prefix+=i.charAt(index);
                if(i.charAt(index) != prefix.charAt(index)) return prefix.substring(0,index);
            }
            index++;
        }
        return "";
    }
}

File: leetcode/longest_common_prefix.py
# https://leetcode.com/problems/longest-common-prefix


class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        prefix: str = ""

        for i in range(len(strs[0])):
            for j in range(1, len(strs)):
                if i >= len(strs[j]) or strs[0][i] != strs[j][i]:
                    return prefix
            prefix += strs[0][i]

        return prefix
